src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,281 @@
-                workspace will be used.
+            client: the `httpx.Client` initialized to interact with the Argilla API.
+            name: the name of the workspace to be managed.
             id: the ID of the workspace to be managed.
+            inserted_at: the datetime when the workspace was created.
+            updated_at: the datetime when the workspace was last updated.
+
+        Raises:
+            ValueError: if the workspace does not exist in the current Argilla account.
+            ValidationApiError: if the ID provided is not a valid UUID.
+            RuntimeError: if there was an error while retrieving the workspace/s from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace(
+            ...     client=rg.active_client().http_client.httpx,
+            ...     name="my-workspace",
+            ...     id="...",
+            ...     inserted_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ...     updated_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ... )
         """
+        self.client = client
+        self.id = id if isinstance(id, UUID) else UUID(id)
         self.name = name
-        self.id = id or None
+        self.inserted_at = inserted_at
+        self.updated_at = updated_at
 
-        if self.id is None:
-            warnings.warn(
-                f"The `rg.Workspace` doesn't exist yet, you need to create it in advance. Otherwise, use `rg.Workspace.from_name` to make sure that the workspace exists."
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(self.client, self.id).parsed
+        return self.__users
+
+    def __str__(self) -> str:
+        return (
+            f"Workspace(id={self.id}, name={self.name}, users={self.users}, insterted_at={self.inserted_at},"
+            f" updated_at={self.updated_at}"
+        )
+
+    # TODO(alvarobartt): also allow user addition via name (which is more user-friendly)
+    def add_user(self, id: str) -> None:
+        """Adds an existing user to the workspace in Argilla.
+
+        Args:
+            id: the ID of the user to be added to the workspace. The user must exist in Argilla.
+
+        Raises:
+            ValueError: if the user with the provided ID already exists in the workspace.
+            RuntimeError: if there was an error while adding the user to the workspace.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace.from_name("my-workspace")
+            >>> workspace.add_user("my-user-id")
+        """
+        try:
+            created_user = workspaces_api.create_workspace_user(
```

Comment:
This is what we had before, a validation layer against Argilla in the `__init__`, but since we decided to remove it, I think we can just rely on the API validation included as part of those endpoints

Reply:
Either calling the API to check whether the user exists or not and then to add it to the workspace (2 API calls) or just add it to the workspace (1 API call) and if it fails we handle the exception to show either that the user doesn't exist and/or the user is already there, in case it fails.